src/WebJobs.Extensions.Http/Extensions/HttpRequestExtensions.cs

```diff
@@ -63,25 +64,36 @@ public static ClaimsIdentity GetAppServiceIdentity(this HttpRequest request)
                 return null;
             }
             string headerValue = request.Headers[EasyAuthIdentityHeader].First();
-            return FromBase64EncodedJson(headerValue);
+            return FromBase64EncodedJson<ClaimsIdentitySlim>(headerValue, EasyAuthClaimsIdentitySerializer.Value);
         }
 
-        private static ClaimsIdentity FromBase64EncodedJson(string payload)
+        public static ClaimsIdentity GetStaticWebAppsIdentity(this HttpRequest request)
```

Comment:
So it looks like both this and the original `GetAppServiceIdentity()` throw an exception if the "x-ms-client-principal" header doesn't match the expected format. Since the format can now feasibly be either, that means that catching the exception will be forced to be part of the normal control flow, which is not ideal. 

My recommendation would be to do one of the following:

1. Make these `TryGet*` method so that it is more apparent that this may not successfully return the expected ClaimsIdentity.
2. Return a `null` value. This generally can lead to more aesthetically pleasing code (i.e. `var headerIdentity = httpContext.GetAppServiceIdentity() ?? httpContext.GetStaticWebAppsIdentity()`), but if you go with this approach, you have to properly document that the methods can return null, and still be aware that some consumers of the method will not read the method docs and won't properly handle the null case.

I would probably recommend solution 1.